types/lodash/ts3.1/common/array.d.ts

```diff
@@ -0,0 +1,2090 @@
+import _ = require("../index");
+declare module "../index" {
+    interface LoDashStatic {
+        /**
+         * Creates an array of elements split into groups the length of size. If collection can’t be split evenly, the
+         * final chunk will be the remaining elements.
+         *
+         * @param array The array to process.
+         * @param size The length of each chunk.
+         * @return Returns the new array containing chunks.
+         */
+        chunk<T>(array: List<T> | null | undefined, size?: number): T[][];
+    }
+    interface Collection<T> {
```

Comment:
I'm not a huge fan of some of these names (`Collection<T>`,  `Object<T>`) because they sound like other types. MY C#/Java brain looks at `Collection<T>` and thinks it's a lot like `List<T>` or `Array<T>`, but it's not. That said I can't think of a better name that's not painfully long.

Reply:
I'd like to use `List`, but got a name conflict with `type List<T> = ArrayLike<T>`. Maybe if I change it to `interface List<T> extends ArrayLike<T> { }`.

Reply:
Later: I tried that and compilation never finished.

Other ideas are welcome!

Reply:
Collection is probably fine. I was thinking something like `CollectionWrapper` or `ListWrapper` (and `ObjectWrapper`, `PrimitiveWrapper`, etc) to indicate that they aren't real collecttions, they are just wrappers. I don't feel too strongly though.